src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/WebJobsTelemetryInitializer.cs

```diff
@@ -26,36 +28,75 @@ public void Initialize(ITelemetry telemetry)
 
             telemetry.Context.Cloud.RoleInstance = _roleInstanceName;
 
+            RequestTelemetry request = telemetry as RequestTelemetry;
+
             // Zero out all IP addresses other than Requests
-            if (!(telemetry is RequestTelemetry))
+            if (request == null)
             {
                 telemetry.Context.Location.Ip = LoggingConstants.ZeroIpAddress;
             }
+            else
+            {
+                if (request.Context.Location.Ip == null)
+                {
+                    request.Context.Location.Ip = LoggingConstants.ZeroIpAddress;
+                }
+            }
 
             // Apply our special scope properties
-            IDictionary<string, object> scopeProps = DictionaryLoggerScope.GetMergedStateDictionary() ?? new Dictionary<string, object>();
-
-            telemetry.Context.Operation.Name = scopeProps.GetValueOrDefault<string>(ScopeKeys.FunctionName);
+            IDictionary<string, object> scopeProps =
+                DictionaryLoggerScope.GetMergedStateDictionary() ?? new Dictionary<string, object>();
 
-            // Apply Category and LogLevel to all telemetry
-            if (telemetry is ISupportProperties telemetryProps)
+            // this could be telemetry tracked in scope of function call - then we should apply the logger scope
+            // or RequestTelemetry tracked by the WebJobs SDK or AppInsight SDK - then we should apply Activity.Tags
+            if (request == null && scopeProps.Any())
```

Comment:
How are you determining whether the `RequestTelemetry` is part of a function call here? Shouldn't `RequestTelemetry` not be null in that case?

Reply:
RequestTelemetry is never null (initializer runs on tracked telemetry item). The question is: who reported RequestTelemetry: functions in LogFunctionResult or AppInsights SDK automagically